src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -202,7 +202,22 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     }
 
     if (this._config?.geo_location_sources) {
-      if (oldHass.states !== this.hass.states) {
+      const oldSourceEntities = this._getSourceEntities(
+        oldHass.states,
+        this._config
+      );
+      const newSourceEntities = this._getSourceEntities(
+        this.hass.states,
+        this._config
+      );
+
+      if (
+        !deepEqual(oldSourceEntities, newSourceEntities) ||
```

Comment:
The render function of this element is really light, so it is not a big deal to call that often, we should just make sure we don't update `_mapEntities` too often.

Reply:
Can you please elaborate? Should `if (this._config?.geo_location_sources)` block always return true (without checking anything else)?
Otherwise, we need the `deepEqual` to trigger the call to `willUpdate`.

Reply:
Done